src/metabase/lib/metadata/calculation.cljc

```diff
@@ -450,6 +450,11 @@
   [_query _stage-number _x _options]
   [])
 
+;;; if you pass in an integer assume it's a stage number; use the method for the query stage itself.
+(defmethod returned-columns-method :dispatch-type/integer
```

Comment:
This is a convenience so you can more easily use the new walk stuff that applies functions to a specific path e.g.

```clj
(lib.walk/apply-f-for-stage-at-path lib/returned-columns query path)
```

(more on this below)